src/cmap/connect.ts

```diff
@@ -200,11 +200,12 @@ export async function prepareHandshakeDocument(
   const options = authContext.options;
   const compressors = options.compressors ? options.compressors : [];
   const { serverApi } = authContext.connection;
+  const clientMetadata = await addContainerMetadata(options.metadata);
```

Comment:
Currently, we cache the result of checking for a `.dockerenv` file (i.e., whether or not we're running in a docker environment) but we don't cache the handshake's client metadata object.  Consequently, every connection attempt will attempt to add the environment metadata (looping over keys in the metadata document, calling `ifItFitsItSits` for each key).  

I was imagining that, instead of continuing to build the metadata document during options parsing and then adding the env metadata at connect time, we'd instead make the entirety of metadata document construction async, build the whole client metadata object during connect time but cache the result so we only calculate it once per-client.  ([durran's suggested approach](https://mongodb.slack.com/archives/GGWBN4ZNK/p1709068406115619) is another way to achieve the outcome of only building the env metadata once).

thoughts?

Reply:
Caching the entire metadata object is a good idea, I'm good to implement this and store it as a module scoped variable.